crest/Assets/Crest/Crest/Scripts/Underwater/UnderwaterRenderer.cs

```diff
@@ -0,0 +1,165 @@
+    [HelpURL(Internal.Constants.HELP_URL_BASE_USER + "underwater.html" + Internal.Constants.HELP_URL_RP)]
+    public partial class UnderwaterRenderer : MonoBehaviour
+    {
+        /// <summary>
+        /// The version of this asset. Can be used to migrate across versions. This value should
+        /// only be changed when the editor upgrades the version.
+        /// </summary>
+        [SerializeField, HideInInspector]
+#pragma warning disable 414
+        int _version = 0;
+#pragma warning restore 414
+
+        // This adds an offset to the cascade index when sampling ocean data, in effect smoothing/blurring it. Default
+        // to shifting the maximum amount (shift from lod 0 to penultimate lod - dont use last lod as it cross-fades
+        // data in/out), as more filtering was better in testing.
+        [Range(0, LodDataMgr.MAX_LOD_COUNT - 2)]
+        [Tooltip("How much to smooth ocean data such as water depth, light scattering, shadowing. Helps to smooth flickering that can occur under camera motion.")]
+        public int filterOceanData = LodDataMgr.MAX_LOD_COUNT - 2;
+
+        [Tooltip("Add a meniscus to the boundary between water and air.")]
+        public bool meniscus = true;
+
+
+        [Header("Advanced")]
+
+        [Tooltip("Copying params each frame ensures underwater appearance stays consistent with ocean material params. Has a small overhead so should be disabled if not needed.")]
+        public bool copyOceanMaterialParamsEachFrame = true;
+
+        [Predicated("useHorizonSafetyMarginMultiplier", inverted: true), Range(0f, 1f)]
+        [Tooltip("Adjusts the far plane for horizon line calculation. Helps with horizon line issue. (Experimental)")]
+        public float farPlaneMultiplier = 0.68f;
+
+        [Tooltip("Use the old horizon safety margin multiplier to fix horizon line issues instead of the new experimental far plane multiplier.")]
+        public bool useHorizonSafetyMarginMultiplier = false;
+
+        // A magic number found after a small-amount of iteration that is used to deal with horizon-line floating-point
+        // issues. It allows us to give it a small *nudge* in the right direction based on whether the camera is above
+        // or below the horizon line itself already.
+        [Predicated("useHorizonSafetyMarginMultiplier"), Range(0f, 1f)]
+        [Tooltip("A safety margin multiplier to adjust horizon line based on camera position to avoid minor artifacts caused by floating point precision issues, the default value has been chosen based on careful experimentation.")]
+        public float horizonSafetyMarginMultiplier = 0.01f;
+
+        [Predicated("useHorizonSafetyMarginMultiplier"), DecoratedField]
+        [Tooltip("Dynamic resolution can cause the horizon gap issue to widen. Scales the safety margin multiplier to compensate.")]
+        public bool scaleSafetyMarginWithDynamicResolution = true;
+
+        [Space(10)]
+        public DebugFields debug = new DebugFields();
```

Comment:
'_' missing

Reply:
It's public. Aren't "_" for private members?

Reply:
no the underscore is for non-static members. properties have no underscore, but members do

Reply:
Sorry. Looks like I need to brush up on the contributing document again... should I make them all private too? I don't think anyone would really need them public.

Reply:
if in doubt, private is a good default choice i reckon. i feel theres no point creating serialised data (that might need to be managed/migrated/fixed in the future) if we're not sure.